feat: Add useScriptLoader custom hook (#478)
Shopify/hydrogen: packages/hydrogen/src/hooks/useLoadScript/docs/2-return-value.md

## Return value

> The `useLoadScript` hook returns a string of three potential values, that allow you to understand the state of the external script you are loading:

Reviewer: Just future-proofing in the case that we offer more than three values in the future! 

```suggestion
The `useLoadScript` hook returns the following values that allow you to understand the state of the external script you are loading:
```

| Value     | Description                                                                                                                              |
| --------- | ---------------------------------------------------------------------------------------------------------------------------------------- |

> | `loading` | The script is still loading. Note that the script tag could be on the page but the resource has not fully loaded yet while in this state |
> | `done`    | The script is fully loaded and ready to use                                                                                              |
> | `error`   | There was an error loading the script                                                                                                    |

Reviewer: ```suggestion
| `loading` | The script is still loading. For example, the script tag can be on the page but the resource might not be fully loaded yet while in this state. |
| `done`    | The script is fully loaded and ready to use.                                                                                              |
| `error`   | There was an error loading the script.                                                                                                   |
```
